File: examples/jmediator-example-handlers/src/main/java/jmediator/example/handlers/ExampleQueryHandler.java
package jmediator.example.handlers;

import jmediator.example.ExampleQuery;
import jmediator.example.ExampleQueryResponse;
import jmediator.MessageHandler;
import reactor.core.publisher.Mono;

public class ExampleQueryHandler implements MessageHandler<ExampleQuery, ExampleQueryResponse> {
    @Override
    public Mono<ExampleQueryResponse> handle(ExampleQuery message) {
        return Mono.just(new ExampleQueryResponse(){{
            setResult("Result for: " + message.getFilter());
        }});
    }
}

File: examples/jmediator-example-handlers-kotlin/src/main/kotlin/jmediator/example/handlers/ExampleQueryHandler.kt
package jmediator.example.handlers

import jmediator.example.ExampleQuery
import jmediator.example.ExampleQueryResponse
import jmediator.toMono
import jmediator.MessageHandler
import reactor.core.publisher.Mono

class ExampleQueryHandler : MessageHandler<ExampleQuery, ExampleQueryResponse> {
    override fun handle(message: ExampleQuery): Mono<ExampleQueryResponse> =
            ExampleQueryResponse().apply {
                result = "(kotlin)Result for: ${message.filter}"
            }.toMono()
}
